src/app/model/Constances.ts

```diff
@@ -1 +1,77 @@
-export const BOARDSIZE = 8;
+import { Side } from './Types';
+import { Rook } from './Rook';
+import { Knight } from './Knight';
+import { Bishop } from './Bishop';
+import { Queen } from './Queen';
+import { Pawn } from './Pawn';
+
+export const BOARD_SIZE = 8;
+
+export const PIECE_START_POSITION = {
+  A1: new Rook(Side.WHITE),
+  B1: new Knight(Side.WHITE),
+  C1: new Bishop(Side.WHITE),
+  D1: new Queen(Side.WHITE),
+  // E1: new King(Side.WHITE),
+  F1: new Bishop(Side.WHITE),
+  G1: new Knight(Side.WHITE),
+  H1: new Rook(Side.WHITE),
+  A2: new Pawn(Side.WHITE),
+  B2: new Pawn(Side.WHITE),
+  C2: new Pawn(Side.WHITE),
+  D2: new Pawn(Side.WHITE),
+  E2: new Pawn(Side.WHITE),
+  F2: new Pawn(Side.WHITE),
+  G2: new Pawn(Side.WHITE),
+  H2: new Pawn(Side.WHITE),
+  A8: new Rook(Side.BLACK),
+  B8: new Knight(Side.BLACK),
+  C8: new Bishop(Side.BLACK),
+  D8: new Queen(Side.BLACK),
+  // E8: new King(Side.BLACK),
+  F8: new Bishop(Side.BLACK),
+  G8: new Knight(Side.BLACK),
+  H8: new Rook(Side.BLACK),
+  A7: new Pawn(Side.BLACK),
+  B7: new Pawn(Side.BLACK),
+  C7: new Pawn(Side.BLACK),
+  D7: new Pawn(Side.BLACK),
+  E7: new Pawn(Side.BLACK),
+  F7: new Pawn(Side.BLACK),
+  G7: new Pawn(Side.BLACK),
+  H7: new Pawn(Side.BLACK),
+};
+// export const PIECESTARTPOSITION = {
+//   A1: { name: 'Rook', side: Side.WHITE },
+//   B1: { name: 'Knight', side: Side.WHITE },
+//   C1: { name: 'Bishop', side: Side.WHITE },
+//   D1: { name: 'Queen', side: Side.WHITE },
+//   E1: { name: 'King', side: Side.WHITE },
+//   F1: { name: 'Bishop', side: Side.WHITE },
+//   G1: { name: 'Knight', side: Side.WHITE },
+//   H1: { name: 'Rook', side: Side.WHITE },
+//   A2: { name: 'Pawn', side: Side.WHITE },
+//   B2: { name: 'Pawn', side: Side.WHITE },
+//   C2: { name: 'Pawn', side: Side.WHITE },
+//   D2: { name: 'Pawn', side: Side.WHITE },
+//   E2: { name: 'Pawn', side: Side.WHITE },
+//   F2: { name: 'Pawn', side: Side.WHITE },
+//   G2: { name: 'Pawn', side: Side.WHITE },
+//   H2: { name: 'Pawn', side: Side.WHITE },
+//   A8: { name: 'Rook', side: Side.BLACK },
+//   B8: { name: 'Knight', side: Side.BLACK },
+//   C8: { name: 'Bishop', side: Side.BLACK },
+//   D8: { name: 'Queen', side: Side.BLACK },
+//   E8: { name: 'King', side: Side.BLACK },
+//   F8: { name: 'Bishop', side: Side.BLACK },
+//   G8: { name: 'Knight', side: Side.BLACK },
+//   H8: { name: 'Rook', side: Side.BLACK },
+//   A7: { name: 'Pawn', side: Side.BLACK },
+//   B7: { name: 'Pawn', side: Side.BLACK },
+//   C7: { name: 'Pawn', side: Side.BLACK },
+//   D7: { name: 'Pawn', side: Side.BLACK },
+//   E7: { name: 'Pawn', side: Side.BLACK },
+//   F7: { name: 'Pawn', side: Side.BLACK },
+//   G7: { name: 'Pawn', side: Side.BLACK },
+//   H7: { name: 'Pawn', side: Side.BLACK },
```

Comment:
@nowakprojects zmieniłem 
`H7: { name: 'Pawn', side: Side.BLACK }`
na 
`H7: new Pawn(Side.BLACK),`, żeby był typ SquareWithPiece, ale nie wiem czy to jest dobre, bo z kolei później w WebChessView musze zaimportować typ SquareWithPiece, który pochodzi z model/types....
Czyli co zmienić na wersję, która jest zakomentowana?

Reply:
Jak dla mnie po prostu w view zrób nowy typ, w stylu PieceView albo po prostu Piece, który będzie miał przykładowo `{ name: 'Pawn', side: Side.BLACK }`. Wtedy nie masz odniesnia wlasnie do modelu. A presenter za to podstawi to co dostanie z boarda.

Reply:
o tym wlaśnie myślałem, ale jakoś dużo tych typów i do jednej pierdoły robić osobny typ .... dlatego stwierdziłem, że zapytam najpierw

Reply:
![image](https://user-images.githubusercontent.com/31566345/107427821-1462a000-6b22-11eb-8410-2a7fff0dcdaf.png)
ale w sumie ta metoda jest wykorzystywana w presenterze, a należy do View i muszę jej podać jakiego typu przyjmuje argument - czyli dalej mamy odwołanie do typu, który jest w model/Types